vars/stageDeployToOpenshift.groovy

```diff
@@ -12,6 +12,27 @@ def call(def context) {
       destTag: "latest",
       namespace: context.targetProject
     )
+
+    def ocpLatestImage = sh(returnStdout: true, script:"oc get istag ${context.componentId}:latest -n ${context.targetProject} --no-headers", label : "find last image").trim().split(/\s+/)
+    def ocpDockerLatestImageRef = ocpLatestImage[1]
+      
+    def ocpDeployment
+    while(true) {
+      ocpDeployment = sh(returnStdout: true, script:"sleep 10 && oc describe dc ${context.componentId} -n ${context.targetProject} | grep -e ${ocpDockerLatestImageRef} -e Status -e 'Latest Version'", label : "find new deployment").trim().split(/\s+/)
+      
+      echo ("Found last deployment id: ${ocpDeployment[2]} - status ${ocpDeployment[6]}")
+      if (ocpDeployment[6] != "Pending" && ocpDeployment[6] != "Running") {
+          echo "OCP Deployment done - reporting status"
+          break
+      }
+    }
+    
+    if (ocpDeployment[6] != "Complete") {
+      error "Deployment ${context.componentId}${ocpDeployment[2]} failed (status: ${ocpDeployment[6]}), please check the error in the OCP console"
+    }
+    
+    deploymentId = "${context.componentId}-${ocpDeployment[2]}"
+    context.addArtifactURI("OCP Deployment Id", deploymentId)
```

Comment:
I think something like this could work:
```
$ oc rollout status -n foo-dev dc/be-main --watch
replication controller "be-main-589" successfully rolled out
```
It returns the ID you are looking for, and the `--watch` part avoids the need to use `sleep`.

You probably also want to use some kind of timeout - otherwise we're waiting for a long time if the container is crashing ...

Reply:
```
utschig@WKS0000627058 MINGW64 /c/code_bix/opendevstack_at_BIX/ods-jenkins-shared-library (feature/97)
$ oc get build --sort-by=.status.startTimestamp -o jsonpath='{.items[.items.length()-1].metadata.name}' -n pltfmdev-dev -l buildconfig=app-be-adeu
error: error parsing jsonpath {.items[.items.length()-1].metadata.name}, invalid array index .items.length()-1

utschig@WKS0000627058 MINGW64 /c/code_bix/opendevstack_at_BIX/ods-jenkins-shared-library (feature/97)
$ oc get build --sort-by=.status.startTimestamp -o jsonpath='{.items[.items.length-1].metadata.name}' -n pltfmdev-dev -l buildconfig=app-be-adeu
error: error parsing jsonpath {.items[.items.length-1].metadata.name}, invalid array index .items.length-1
```

yak .. at least I have to keep this .. but I'll surround it with a timeout

Reply:
I think this should actually work, just different syntax - see https://support.smartbear.com/alertsite/docs/monitors/api/endpoint/jsonpath.html.